plugin/types.go

```diff
@@ -1,14 +1,39 @@
+/*
+Package plugin defines a set of interfaces that plugins must implement to enable wash
+functonality.
+
+All resources must implement the Entry interface. To do so they should include the EntryBase
+type, and initialize it via NewEntry. For example
+	type myResource struct {
+		plugin.EntryBase
+	}
+	...
+	rsc := myResource{plugin.NewEntry("a resource")}
+EntryBase gives the resource a name - which is how it will be displayed in the filesystem
+or referenced via the API - and tools for controlling how its data are cached.
```

Comment:
`are cached` => `is cached`

Reply:
It might be better if this was in a `docs/core_plugins/` section so we can add more detail and maybe even a walkthrough.

Reply:
I thought it likely people would end up at https://godoc.org/github.com/puppetlabs/wash/plugin while implementing a plugin, and wanted to have some sort of roadmap. We'll also want something more complete walking through how to create a plugin.